lib/rouge/lexers/markdown.rb

```diff
@@ -94,15 +94,15 @@ def html
           (\s*) # leading whitespace
           (\[) (#{edot}+?) (\]) # the reference
           (\s*) (:) # colon
+          (\s*) # trailing whitespace
         )x do
-          groups Text, Punctuation, Str::Symbol, Punctuation, Text, Punctuation
-
-          push :title
+          groups Text, Punctuation, Str::Symbol, Punctuation, Text, Punctuation, Text
+          push :ref
           push :url
         end
 
         # links and images
-        rule %r/(!?\[)(#{edot}*?|[^\]]*?)(\])(?=[\[(])/ do
+        rule %r/(!?\[)((?:\\.|[^\\\]]|`.*?`)*?)(\])(?=[\[(])/ do
```

Comment:
@Ravlen This is the bit I'm unhappy about. It feels like a gross hack to allow the use of `[[` and `]]` for TOML headings but I couldn't see another way to do it.

Reply:
@pyrmont FYI, don't feel like you have to merge this right away, I'm just swamped at work. When I get some free time I fully intend to come back to this and see if I can come up with an alternative. 🙇

Reply:
🤔  As someone coming to this *way* late, I'm not quite sure what the concern is about the TOML headings. @Ravlen any insights? Is the concern it's syntactically incorrect, or the Regex is nasty?